app/models/host/managed.rb

```diff
@@ -653,6 +654,22 @@ def clone
     host
   end
 
+  def check_interfaces
+    primary_flag = false
+    provision_flag = false
+    self.interfaces.each do |interface|
+      if interface.primary
+        primary_flag = true
+      end
+      if interface.provision
+        provision_flag = true
+      end
+    end
+    if !provision_flag || !primary_flag
+      errors.add(:base, _("Some interfaces need to be marked as primary and provision interface"))
```

Comment:
An interface marked as primary is missing / An interface marked as provision is missing

Reads better to me. These should be reported separately:

    errors.add(:base, _("An interface marked as primary is missing")) if interfaces.detect(:&primary)